api/index.js

```diff
@@ -1,42 +1,53 @@
 "use strict";
 
+const Hoek = require(`hoek`);
+
 exports.register = function api(server, options, next) {
+  if (server.app.cacheEnabled) {
+    const catbox = server.root._caches._default.client;
+
+    Hoek.assert(catbox && typeof catbox === `object`, `Can't find catbox cache client`);
+
+    const catboxEngine = catbox.connection;
+
+    Hoek.assert(catboxEngine && typeof catboxEngine === `object`, `Can't find catbox engine`);
+    Hoek.assert(typeof catboxEngine.isReady === `function`, `Catbox engine doesn't have a ready function`);
+  }
+
   // Add each module's cache functions to the global server methods
   [
-    require(`./modules/users/cache`)
+    require(`./modules/users/cache`),
+    require(`./modules/files/cache`)
   ].forEach(module => {
     Object.keys(module).forEach(CacheClassKey => {
-      const Cache = module[CacheClassKey];
+      const cache = new module[CacheClassKey](server);
       let cacheConfig;
 
-      if (process.env.REDIS_URL) {
+      if (server.app.cacheEnabled && cache.config) {
         cacheConfig = {
-          cache: Cache.config
+          cache: cache.config
         };
+
+        if (cache.generateKey) {
+          cacheConfig.generateKey = cache.generateKey;
+        }
       }
 
-      server.method(`cache.${Cache.name}`, Cache.run, cacheConfig);
+      server.method(cache.name, cache.run.bind(cache), cacheConfig);
```

Comment:
it is a prototype function of cache that relies on `this`. Since the bind is a one time cost only (when the server starts), there isn't much of a perf hit. Ideally with ES6 this would just be `(...args) => cache.run(...args)` but node 4 won't let us do that.